src/lib/controllers/UsersController.js

```diff
@@ -20,9 +20,19 @@ class UsersController extends Controller {
   @route('/v1/users')
   @anonymous()
   async createUser(userCredentials: UserCredentials) {
-    // todo add checking for existing usernames.
-    const newUser = await this._usersRepository.create(userCredentials);
-    return this.ok(newUser);
+    try {
+      const isUserNameExist = this._usersRepository.getAll()
```

Comment:
Also, you should move this logic into the repository.  `isUserNameInUse`
That way this can be replaced with some API.